zypp/base/String.h

```diff
@@ -142,10 +142,10 @@ namespace zypp
 #endif
 
     inline std::string asString( const char * t )
-    { return t; }
+    { return t == nullptr ? std::string() : t; }
```

Comment:
Good catch. The changes here are not related to the issue. It's a separate 'bug' so it should go into a separate commit (and we will pick it for the maintenance branches as well).